src/components/app.jsx

```diff
@@ -34,13 +34,17 @@ const PublicContext = withClient(({ client }) => {
   const [sharedDocumentId, setSharedDocumentId] = useState(null)
   const returnUrl = useMemo(() => getReturnUrl(), [])
 
-  useEffect(
-    () =>
-      getSharedDocument(client)
-        .then(id => setSharedDocumentId(id))
-        .catch(() => setSharedDocumentId(false)),
-    []
-  )
+  useEffect(() => {
+    const fetchData = async () => {
```

Comment:
If you want to use a function then call it, it may be clearer if you define `fetchData` outside of `useEffect`

Reply:
"Warning: useEffect function must return a cleanup function or nothing. Promises and useEffect(async () => ...) are not supported, but you can call an async function inside an effect.. "

Reply:
https://stackoverflow.com/a/53332372

Reply:
Oh… mais oui, effectivement. Bon, il aurait suffit de mettre des accolades autour, pas forcément besoin de découper avec une fonction + un appel, mais ça me va ainsi.

Reply:
Bah en fait, c'est la façon de faire un peu standard. On a ça dans à peu près tous les hooks que j'ai vu chez cozy et c'est présenté ainsi sur la doc de React. Du coup je garde la consistance (et puis j'aime pas trop voir trainer des .then / .catch() aussi, mais ça c'est personnel ^^)